components/script/dom/screen.rs

```diff
@@ -26,9 +30,39 @@ impl Screen {
                            window,
                            ScreenBinding::Wrap)
     }
+
+    fn screen_size(&self) -> Size2D<u32> {
+        let (send, recv) = ipc::channel::<(Size2D<u32>)>().unwrap();
+        self.global().script_to_constellation_chan().send(ScriptMsg::GetScreenSize(send)).unwrap();
+        recv.recv().unwrap_or((Size2D::zero()))
+    }
 }
 
 impl ScreenMethods for Screen {
+    // https://drafts.csswg.org/cssom-view/#dom-screen-availwidth
+    fn AvailWidth(&self) -> Finite<f64> {
+        // FIXME: Not spec complient because glutin don't have corresponding API
+        // Fallback to window.screen.width behavior
+        self.Width()
+    }
+
+    // https://drafts.csswg.org/cssom-view/#dom-screen-availheight
+    fn AvailHeight(&self) -> Finite<f64> {
+        // FIXME: Not spec complient because glutin don't have corresponding API
+        // Fallback to window.screen.height behavior
+        self.Height()
```

Comment:
The FIXME should be in `/ports/`, not in servo. Code in `/components/` are backend independent.

Here you should call `self.avail_screen_size().height`, and implement the associated message (`GetAvailScreenSize`).

In `ports/glutin/window.rs`, implement `fn avail_screen_size()`, and there, add the FIXME and use `glutin::get_primary_monitor().get_dimensions();`.